lib/oli_web/live/products/products_view.ex

```diff
@@ -147,6 +163,42 @@ defmodule OliWeb.Products.ProductsView do
     """
   end
 
+  def handle_event("include_archived", __params, socket) do
+    project_id = if socket.assigns.project === nil, do: nil, else: socket.assigns.project.id
+
+    include_archived = !socket.assigns.include_archived
+
+    options = %BlueprintBrowseOptions{
+      project_id: project_id,
+      include_archived: include_archived
+    }
+    products = Blueprint.list(options)
+
+    total_count = length(products)
+
+    {:ok, table_model} = OliWeb.Products.ProductsTableModel.new(products, socket.assigns.ctx)
+
+    socket = assign(socket,
+      include_archived: include_archived,
+      products: products,
+      total_count: total_count,
+      table_model: table_model
+    )
+
+    {:noreply,
+     push_patch(socket,
```

Comment:
It doesn't make sense to call `push_patch` here unless you've implemented also the `handle_params`.  We are going to file another ticket to convert this view over to a fully DB driven table with URL param support, so for this PR, just change this method to return `{:noreply, socket}` after you've updated the socket assigns.

Reply:
`handle_params` is implemented in the `use OliWeb.Common.SortableTable.TableHandlers`, which extends the `product_view`. This implementation is designed to work well (play well) with the other "filtering" conditions. i.e. you can filter and either include or exclude `:archived` products as part of the filter. If not that nuance gets lost

Reply:
`handle_params` is not able to handle "include_archived" - so the issue is that, despite the fact that "push_path" is being called on update to "include_archived", it isn't pushed as a URL param and handled.  To see this, do this:

1. Go into products view. 
2. Toggle "Include archived" (see the archived results appear)
3. Directly navigate away from page to another page. 
4. Use browser back button to return. 

Expected: Include Archived (since it was a path param) is preserved
Actual: it is not

It probably is okay that it is not beign handled, there is another ticket to redo that aspect of this view.  But since we aren't handling it with "handle_params" - we shouldn't push it as a path param.

Reply:
I have tried these same steps on several browsers. Navigating back via the back button works fine no matter where else I navigate. `include_archived` state is correctly preserved and processed. The reason for this is that it is maintained on the mount calls as well.